boards/arm/frdm_k64f/frdm_k64f_defconfig

```diff
@@ -10,3 +10,6 @@ CONFIG_GPIO=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=120000000
 CONFIG_OSC_EXTERNAL=y
 CONFIG_HAS_DTS=y
+
+# Enable MPU
+CONFIG_HAS_SYSMPU=y
```

Comment:
First of all thanks for the info, I was not aware of it. This is definitely something that we need to target in the near future and I am going to send to you a separate email so we can formalize the requirements. 

Said that the patch you are referring to with your changes https://github.com/zephyrproject-rtos/zephyr/commit/5951e45580c198c11e8fac59544195123a46e3ad has been in RFC for nearly 2 months and I was hoping to have captured all the requirements in terms of exceptions but it seems something it is still missing :-)

With respect to the NXP MPU configuration that is part of this specific patch we are not targeting the scenarios that you are describing hence in my previous comment I was suggesting that at this stage without specific requirements the way to use mcuboot like applications is overriding CONFIG_HAS_SYSMPU in the application configuration setting it to n.

The support of the scenarios you described will be part of a future patch once the scope and the usecases are formalized.